packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -878,6 +906,39 @@ export const useAzureCommunicationCallAdapter = (
   return adapter;
 };
 
+const createAzureCommunicationCallAdapterFromClientStable = async (
+  callClient: StatefulCallClient,
+  callAgent: CallAgent,
+  locator: CallAdapterLocator
+): Promise<CallAdapter> => {
+  const deviceManager = (await callClient.getDeviceManager()) as StatefulDeviceManager;
+  return new AzureCommunicationCallAdapter(callClient, locator, callAgent, deviceManager);
+};
+
+let createAzureCommunicationCallAdapterFromClientToExport: (
+  callClient: StatefulCallClient,
+  callAgent: CallAgent,
+  locator: CallAdapterLocator,
+  /* @conditional-compile-remove(rooms) */ roleHint?: Role
```

Comment:
in terms of scaling to more future options, do we want to start making this non-flat? e.g.
```ts
createAzureCommunicationCallAdapterFromClient: (
  callClient,
  callAgent,
  locator,
  options: {
    roleHint: Role
  }
) => Promise<CallAdapter>
```

The moment there's optional arg that's positional in the future this could become:
```ts
createAzureCommunicationCallAdapterFromClient(callClinet, callAgent, locator, undefined /*rolehint*/, undefined /*another option*/, thirdOption)
```

Reply:
Heh... Left the same comment on the API diff.

Reply:
Yeah, you and James both have a good point :). Changed.